vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,510 @@
+  summarise(count = n()) |>
+  mutate(delay = as.integer(report_date - reference_date))
+```
+Looking at this data, we can see that there is one case where there is a negative delay, which indicates that the time stamp of the diagnosis update was recorded before the start of the visit.
+This is likely due to a data entry error, we will choose to exclude all negative values delays in our dataset.
+
+Filter out negative delays.
+```{r}
+count_df <- filter(count_df, delay >= 0)
+head(count_df)
+```
+
+We have now generated data in the format that we need to use the `baslinenowcast` package, which requires a long tidy dataframe with incident case counts indexed by reference date and report date.
+For demonstration purposes, we will now swap out the data from the simulated NSSP line-list data with a larger synthetic dataset.
+
+# Pre-processing of larger synthetic dataset
+
+We'll start by loading in the synthetic reporting triangle dataframe, which is also provided as package data.
+```{r}
+syn_nssp_rep_tri_df
+```
+You can see that this larger synthetic dataset has the same format as the one we generated from the line list NSSP data -- with columns for reference date, report date, and counts.
+
+## Exploratory data analysis to identify an appropriate maximum delay
+
+We will start by computing the reporting delay and performing an exploratory data analysis of the delays in this dataset.
+<details><summary>Click to expand code to create plots of the delay distributions </summary>
+```{r}
+long_df <- syn_nssp_rep_tri_df |>
+  mutate(delay = as.integer(report_date - reference_date))
+
+delay_df_t <- long_df |>
+  group_by(reference_date) |>
+  summarise(mean_delay = sum(count * delay) / sum(count))
+
+delay_summary <- long_df |>
+  mutate(mean_delay_overall = sum(count * delay) / sum(count))
+
+avg_delays <- long_df |>
+  group_by(delay) |>
+  summarise(pmf = sum(count) / sum(long_df$count)) |>
+  mutate(cdf = cumsum(pmf))
+
+delay_t <- ggplot(delay_df_t) +
+  geom_line(aes(
+    x = reference_date,
+    y = mean_delay
+  )) +
+  geom_line(
+    data = delay_summary,
+    aes(
+      x = reference_date,
+      y = mean_delay_overall
+    ),
+    linetype = "dashed"
+  ) +
+  xlab("") +
+  ylab("Mean delay") +
+  theme_bw()
+
+cdf_delay <- ggplot(avg_delays) +
+  geom_line(aes(x = delay, y = cdf)) +
+  geom_hline(aes(yintercept = 0.95), linetype = "dashed") +
+  theme_bw()
+```
+</details>
+```{r}
+delay_t
+cdf_delay
+```
+Based on this figure, we can set the maximum delay to be 25 days as this is where 95% of the cases appear to have been reported.
+We will also set the nowcast date as 30 days before the maximum reference date in the dataset, May 5th, 2026.
+
+```{r}
+max_delay <- 25
+nowcast_date <- max(long_df$reference_date) - days(30)
+```
+
+## Format for `baselinenowcast`
+We'll start by generating a reporting triangle from the `syn_nssp_rep_tri_df`, after we've removed all of the reports from after the nowcast date.
```

Comment:
link to this concept

here the df already being called rep tri is pretty confusing with the way the text is written